metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala

```diff
@@ -279,7 +283,43 @@ final class ImplementationProvider(
 
     val allLocations = new ConcurrentLinkedQueue[Location]
 
-    for {
+    def findImplementationLocations(
```

Comment:
@alexarchambault I reworked this part in order not to use parallel collections since they seems to be randomly breaking.

You can see the changes here https://github.com/scalameta/metals/pull/3790/commits/e0305f9e7468947c0a173abfc75acf55b13ff369

Do you have any idea why using global execution context was breaking things? It shouldn't be used for sure, but I am worried about what might have caused it.

Reply:
I don't know for sure, but my guess is that the Scala CLI tests were calling things in a way that could make the global execution context starve. So that the Scala CLI changes were only revealing the underlying issue, and not causing it themselves.

Reply:
Hmm.... that might have been it, thanks! I will take last look today or tomorrow and merge